### ISSUE TITLE
Filmstrip Comparisson View Showing First Test Only

### ISSUE THREAD

**Issue by ErikWitt** [WPO-Foundation/webpagetest#1467]
With the new design it seems that the filmstrip comparison view under https://www.webpagetest.org/ broke. It only shows the information for the first test.

Example:
https://www.webpagetest.org/video/compare.php?tests=210331_BiDcH0_a1afd9bf986c32498dba8e71f4c788a6,210331_AiDcKV_c1c31d6a7e6515dcfc7186eed4ae16c2

**Comment by pmeenan** [comment]
tl;dr: should be better now.

Unrelated to the new UI but it looks like it was an issue from the multi-server deployment yesterday.  If you went to the compare page before the test completed (and made it into the shared store) then it could end up in a state where it tried to pull in a test from the shared store, marked it as missing and then wouldn't try again.

I just pushed an update so that it doesn't mark the test as missing so it can be restored from the shared storage when accessed.

in this case the tests spanned 2 different servers which is what tripped the race (The "A" and "B" right after the date in the test ID's).

Let me know if you see it continue.  In that case there may be an issue with the timeliness in a test result being uploaded to the shared storage (should happen as soon as a test completes).

**Comment by pmeenan** [comment]
Hmm, thanks - looking into it now.  It's possible that published tests aren's uploaded to the archive immediately like tests that are run on the instance are.

**Comment by pmeenan** [comment]
OK, just updated the publish receiver to also archive tests immediately and I triggered an archive on both of those published tests.  New ones should just work but let me know if you have any others that you need me to force-archive

**Comment by pmeenan** [comment]
Or, if you want to trigger archiving for tests you captured without sharing the ID's, you can trigger it youself.  There is an archive.php endpoint that takes a test ID:

https://www.webpagetest.org/archive.php?test=210331_AiDcMQ_87ac2789b90d346fb82a3fe875df0a88

**Comment by ErikWitt** [comment]
Awesome, that works like a charm. Thanks!
